### PR TITLE
[ompl] Fix linking using ${OMPL_LIBRARIES}

### LINKED ISSUE #18237: [ompl] Cannot find lib files when using CMake
**Describe the bug**
After doing `find_package` and successfully finding ompl, linking fails because it can't find `/lib/boost_serialization-vc140-mt.lib`

**Environment**
- OS: Windows 10
- Compiler: MSVC 2019

**To Reproduce**
1. `./vcpkg install ompl --triplet=x64-windows`
2. `find_package(ompl REQUIRED)`

**Expected behavior**
I expect to be able to link the library after successfully using `find_package`

**Failure logs** 
`LINK : fatal error LNK1181: cannot open input file '\lib\boost_serialization-vc140-mt.lib'`

**Additional context**
This appears to occur because both `${OMPL_LIBRARY_DIRS}` and `${OMPL_LIBRARIES}` use `${_IMPORT_PREFIX}` in omplConfig.cmake which seems to be empty, leading to paths like `/lib` and `/debug/lib` being in `${OMPL_LIBRARY_DIRS}` and `/lib/boost_serialization-vc140-mt.lib` in `${OMPL_LIBRARIES}`. During port installation, all the paths were correct absolute paths at one point, and parts of the paths were replaced by `${_IMPORT_PREFIX}` when `vcpkg_fixup_cmake_targets` was called.

### PR DESCRIPTION
**Pull request by Ace314159**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #18237

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM, thanks for your fixing @Ace314159.

**Comment by PhoebeHui** [comment]
Actually, ompl has the following issue:
1. Usage issue: ompl need to export the cmake targets file, in order to avoid linking to release libs when build with debug configuration.
2. Cmake config: need to add the dependency port boost and eigen in omplConfig.cmake file.
3. Need to divide the feature 'omplapp' into seperate port. and let it depend on ompl. since they has different source structure, it's hard to apply patch to them, eg, we patch the omplConfig.cmake file to ompl, however, the when build it with feature 'omplapp', it will fail to apply the patch. I assume the feature 'omplapp' would fail here.

**Comment by Ace314159** [comment]
Sorry for the delay. I'm not that familiar with the cmake targets, so I'm not sure if I did it correctly, but do these changes address the issues?

**Comment by Ace314159** [comment]
The failures with static seem to be because `omlapp` fails to link with boost and ccd. This same issue occurs even when installing the feature directly, before I made it a separate port.

I'm not sure how to fix this. Do you have any suggestions?

**Comment by Ace314159** [comment]
I was able to fix the build issues. Does everything look good now?

File: ports/omplapp/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        -DOMPL_BUILD_PYTESTS=OFF
+)
+
+vcpkg_install_cmake()
+
+vcpkg_fixup_cmake_targets(CONFIG_PATH share/ompl/cmake)
+
+# Remove debug distribution and other, move ompl_benchmark to tools/ dir
+file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/tools/${PORT})
+file(RENAME ${CURRENT_PACKAGES_DIR}/bin/ompl_benchmark.exe ${CURRENT_PACKAGES_DIR}/tools/${PORT}/ompl_benchmark.exe)
```
Comment (PhoebeHui) [inline ports/omplapp/portfile.cmake:49]: vcpkg_copy_tools(TOOL_NAMES ompl_benchmark AUTO_CLEAN)

Comment (PhoebeHui) [inline ports/omplapp/portfile.cmake:49]:
```suggestion
vcpkg_copy_tools(TOOL_NAMES ompl_benchmark AUTO_CLEAN)
```

Comment (PhoebeHui) [inline ports/omplapp/portfile.cmake:49]:
```suggestion
vcpkg_copy_tools(TOOL_NAMES ompl_benchmark AUTO_CLEAN)
```

File: ports/omplapp/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    ${CURRENT_PACKAGES_DIR}/debug/lib/ompl.lib
+    ${CURRENT_PACKAGES_DIR}/debug/share
+)
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
+endif()
+
+# Handle copyright
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (PhoebeHui) [inline ports/omplapp/portfile.cmake:68]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

```

File: ports/omplapp/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    ${CURRENT_PACKAGES_DIR}/include/ompl
+    ${CURRENT_PACKAGES_DIR}/bin
+    ${CURRENT_PACKAGES_DIR}/include/omplapp/CMakeFiles
+    ${CURRENT_PACKAGES_DIR}/lib/ompl.lib
+    ${CURRENT_PACKAGES_DIR}/share/ompl
+    ${CURRENT_PACKAGES_DIR}/share/man
+    ${CURRENT_PACKAGES_DIR}/debug/bin
+    ${CURRENT_PACKAGES_DIR}/debug/include
+    ${CURRENT_PACKAGES_DIR}/debug/lib/ompl.lib
+    ${CURRENT_PACKAGES_DIR}/debug/share
```
Comment (PhoebeHui) [inline ports/omplapp/portfile.cmake:60]: Could you also add qutotes?
See https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/cmake-guidelines.md

**Comment by PhoebeHui** [comment]
Others LGTM.

**Review by PhoebeHui — APPROVED** [review]

**Pull request merged**